File: client/src/app/modules/tiers/services/sales.complain.cause.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { SalesComplainCause } from '../../miscellaneous/model/SalesComplainCause';

@Injectable({
  providedIn: 'root'
})
export class SalesComplainCauseService extends AppRestService<SalesComplainCause>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getComplainCauses() {
    return this.getListCached(new HttpParams(), "sales-complain-causes");
  }

   addOrUpdateComplainCause(salesComplainCause: SalesComplainCause) {
    return this.addOrUpdate(new HttpParams(), "sales-complain-cause", salesComplainCause, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/SalesComplainCauseService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.SalesComplainCause;

public interface SalesComplainCauseService {

        public List<SalesComplainCause> getComplainCauses();

        public SalesComplainCause addOrUpdateComplainCause(SalesComplainCause salesComplainCause);
}
